bug: operator-builder cannot use `make run` target

Lots of errors when attempting to run `make run`:

```
# github.com/banzaicloud/operator-tools/pkg/utils
../../../go/pkg/mod/github.com/banzaicloud/operator-tools@v0.25.4/pkg/utils/log.go:37:5: cannot use logger{...} (type logger) as type logr.Logger in assignment
../../../go/pkg/mod/github.com/banzaicloud/operator-tools@v0.25.4/pkg/utils/log.go:44:9: cannot use &logger{...} (type *logger) as type logr.Logger in return argument
../../../go/pkg/mod/github.com/banzaicloud/operator-tools@v0.25.4/pkg/utils/log.go:76:15: cannot use logger{...} (type logger) as type logr.Logger in return argument
../../../go/pkg/mod/github.com/banzaicloud/operator-tools@v0.25.4/pkg/utils/log.go:87:15: cannot use logger{...} (type logger) as type logr.Logger in return argument
../../../go/pkg/mod/github.com/banzaicloud/operator-tools@v0.25.4/pkg/utils/log.go:98:15: cannot use logger{...} (type logger) as type logr.Logger in return argument
```

@lander2k2 and I pinned this down to the `go.mod` file being auto-updated by `go mod tidy` during each create api event.  We need to hard-pin the versions we want.

--- a/internal/plugins/workload/v1/scaffolds/init.go
+++ b/internal/plugins/workload/v1/scaffolds/init.go
@@ -11,17 +11,11 @@ import (
 	"sigs.k8s.io/kubebuilder/v3/pkg/config"
 	"sigs.k8s.io/kubebuilder/v3/pkg/machinery"
 	"sigs.k8s.io/kubebuilder/v3/pkg/plugins"
-	"sigs.k8s.io/kubebuilder/v3/pkg/plugins/golang/v3/scaffolds"
 
 	"github.com/vmware-tanzu-labs/operator-builder/internal/plugins/workload/v1/scaffolds/templates"
 	"github.com/vmware-tanzu-labs/operator-builder/internal/plugins/workload/v1/scaffolds/templates/cli"
 	"github.com/vmware-tanzu-labs/operator-builder/internal/plugins/workload/v1/scaffolds/templates/test/e2e"
 	workloadv1 "github.com/vmware-tanzu-labs/operator-builder/internal/workload/v1"
-)
-
-const (
-	CobraVersion               = "v1.1.3"
-	OperatorBulderToolsVersion = "v0.2.0"
 )
 
 var _ plugins.Scaffolder = &initScaffolder{}
@@ -83,11 +77,7 @@ func (s *initScaffolder) Scaffold() error {
 
 	if err := scaffold.Execute(
 		&templates.Main{},
-		&templates.GoMod{
-			ControllerRuntimeVersion:    scaffolds.ControllerRuntimeVersion,
-			CobraVersion:                CobraVersion,
-			OperatorBuilderToolsVersion: OperatorBulderToolsVersion,
-		},
+		&templates.GoMod{},
 		&templates.Dockerfile{},
 		&templates.Makefile{RootCmdName: s.cliRootCommandName},
 		&templates.Readme{RootCmdName: s.cliRootCommandName},

--- a/internal/plugins/workload/v1/scaffolds/templates/gomod.go
+++ b/internal/plugins/workload/v1/scaffolds/templates/gomod.go
@@ -14,9 +14,29 @@ type GoMod struct {
 	machinery.TemplateMixin
 	machinery.RepositoryMixin
 
-	ControllerRuntimeVersion    string
-	CobraVersion                string
-	OperatorBuilderToolsVersion string
+	Dependencies map[string]string
+}
+
+// goModDependencyMap pins the versions within the go.mod file so that they
+// do not get auto-updated.
+//
+// See https://github.com/vmware-tanzu-labs/operator-builder/issues/250
+func goModDependencyMap() map[string]string {
+	return map[string]string{
+		"github.com/go-logr/logr":                    "v0.4.0",
+		"github.com/nukleros/operator-builder-tools": "v0.2.0",
+		"github.com/onsi/ginkgo":                     "v1.16.4",
+		"github.com/onsi/gomega":                     "v1.15.0",
+		"github.com/spf13/cobra":                     "v1.2.1",
+		"github.com/stretchr/testify":                "v1.7.0",
+		"gopkg.in/yaml.v2":                           "v2.4.0",
+		"k8s.io/api":                                 "v0.22.2",
+		"k8s.io/apimachinery":                        "v0.22.2",
+		"k8s.io/client-go":                           "v0.22.2",
+		"sigs.k8s.io/controller-runtime":             "v0.10.2",
+		"sigs.k8s.io/kubebuilder/v3":                 "v3.2.0",
+		"sigs.k8s.io/yaml":                           "v1.2.0",
+	}
 }
 
 func (f *GoMod) SetTemplateDefaults() error {
@@ -24,8 +44,8 @@ func (f *GoMod) SetTemplateDefaults() error {
 		f.Path = "go.mod"
 	}
 
+	f.Dependencies = goModDependencyMap()
 	f.TemplateBody = goModTemplate
-
 	f.IfExistsAction = machinery.OverwriteFile
 
 	return nil
@@ -37,9 +57,8 @@ module {{ .Repo }}
 go 1.15
 
 require (
-	sigs.k8s.io/controller-runtime {{ .ControllerRuntimeVersion }}
-	github.com/spf13/cobra {{ .CobraVersion }}
-	github.com/nukleros/operator-builder-tools {{ .OperatorBuilderToolsVersion }}
-
+	{{ range $k, $v := $.Dependencies }}
+	"{{ $k }}" {{ $v }}
+	{{ end -}}
 )
 `

